### PR TITLE
grin: switch to `llvm@15`

### PR DESCRIPTION
**Pull request by ZhongRuoyu**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

A follow-up to #128865. The error '"enum_(unnamed_at_CRoaring/ roaring_h_7_1)" is not a valid Ident' was from `croaring`, caused by rust-lang/rust-bindgen#2312. We could fix this by switching to `llvm@15`.

~~In addition, add `lmdb` as a dependency to avoid vendoring.~~ (Update: see https://github.com/Homebrew/homebrew-core/pull/130508#issuecomment-1539672712.) `croaring` is being vendored too, but the crate `croaring-sys` only supports building its own copy (see [^1]). Given that it's lightweight (with a single C source file), I think we can leave it as-is for now.

[^1]: https://github.com/saulius/croaring-rs/blob/0.8.1/croaring-sys/build.rs

### CODE REVIEW

File: Formula/grin.rb [hunk 1]
```diff
@@ -18,8 +18,14 @@ class Grin < Formula
     sha256 cellar: :any_skip_relocation, x86_64_linux:   "cdcbd7bcb60160cbd90c54e362650a191b64fb33d356991896c0778ced854ecf"
   end
 
-  depends_on "llvm" => :build # for libclang
+  # Use `llvm@15` to work around build failure with Clang 16 described in
+  # https://github.com/rust-lang/rust-bindgen/issues/2312.
+  # TODO: Switch back to `uses_from_macos "llvm" => :build` when `bindgen` is
+  # updated to 0.62.0 or newer. Check:
+  #   https://github.com/mimblewimble/grin/blob/v#{version}/Cargo.lock
```
Comment (carlocab) [inline Formula/grin.rb:25]: Can we parse `Cargo.lock` in `install` and then `odie` when this needs to change?

Comment (ZhongRuoyu) [reply Formula/grin.rb:25]: That's a good idea!

**Review by p-linnane — DISMISSED** [review]

**Comment by carlocab** [comment]
Can we mock a `pc` file and write it to `buildpath/"pkgconfig/lmdb.pc"` or something like that?

**Comment by ZhongRuoyu** [comment]
Yea, I'm currently working to add a `lmdb.pc` file to our `lmdb`. Debian does this too: https://salsa.debian.org/debian/lmdb/-/blob/debian/0.9.24-1/debian/lmdb.pc.in?ref_type=tags

**Review by carlocab — CHANGES_REQUESTED** [review]
Needs rebase

**Comment by ZhongRuoyu** [comment]
Ah. There are two crates: `lmdb-sys` and `liblmdb-sys`. I was looking at the former but `grin` is actually using the latter, a 6-year-old 🙃

https://github.com/mimblewimble/grin/blob/v5.1.2/Cargo.lock#L1467

And unfortunately, `liblmdb-sys` cannot link with system `lmdb` at all.

https://github.com/vhbit/lmdb-rs/blob/master/liblmdb-sys/build.rs

**Review by carlocab — APPROVED** [review]

**Pull request merged**